### DISCUSSION TITLE
TAB key hardware keyboard

### DISCUSSION THREAD

**Discussion by zsh-is-life** [gsantner/markor#2484] [Q&A]
Whenever I wish to make a nested list, I end up switching to the bottom toolbar. I press the "Tab" key on my Bluetooth keyboard, but it only starts to shuffle between the icons in the bottom toolbar. Any suggestions?

Im running Android 15 on a OnePlus Pad 2.

https://github.com/user-attachments/assets/4f4ff38d-cd49-4b9a-bdd4-1178342f4db0

**Comment by harshad1** [comment]
We use spaces for indentation and never added tab => space conversion

Ideally we do both: add tab support for nested lists and automatic conversion.

**Reply by zsh-is-life** [reply]
Strange, I can not get it to work. After the first line in the list, I hit enter, I am presented with the bullet point (as expected). If I press space after that and enter some text, no nested list is formed. If I delete the bullet point, add some space in the very beginning, add a bullet point, add text, I again see no nested list.

Maybe I'm doing something wrong, besides I feel tab and shift+tab are the most intuitive keystrokes for indent/deindent a list - since most popular text editors understand this. Google Docs, MS Word, Typora and countless other text/markdown/docx editors.

**Reply by harshad1** [reply]
To indent the list you need to put the spaces before the bullet / number. 

We don't intercept the tab key yet. I will implement this.

**Comment by harshad1** [comment]
@gsantner we can intercept the tab key and indent the list (shift + tab would deindent).

IMO this would require a global option.

What do you think?

**Reply by gsantner** [reply]
Yes, suggestion for top bar though, with toggle global only not file specfic

**Reply by harshad1** [reply]
> with toggle global only not file specfic

Yeah, agreed

> suggestion for top bar though

Not sure I understand

**Reply by gsantner** [reply]
Where the Auto-Format checkbox toggle is

**Comment by harshad1** [comment]
It looks like default tab behaviour on android is to jump to the next view. Is this useful at all? 

IMO we just change tab to indent by default without needing an option to change that behaviour.

**Reply by gsantner** [reply]
Yes, at least blind or otherwise disabled users it can be very important to be able to navigate this way. I think it's generally good to keep the hardware tab key based or otherwise assisted tab/next UI navigation around.


What do you think @schulle4u? #613

**Reply by schulle4u** [reply]
I don't have a strong opinion on this. Screen readers like TalkBack usually have their own shortcuts for navigating between GUI elements (e.g. Alt+Arrow keys), so a different behavior of the tab key would probably not be disruptive. On the other hand, TalkBack is no longer the only screen reader available for Android, and it could cause problems for users of alternative screen readers. Personally I find tab stops ugly and generally use spaces. 🙂
